catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddBrickFragment.java

```diff
@@ -79,7 +78,12 @@ public static AddBrickFragment newInstance(String selectedCategory, ScriptFragme
 
 	@Override
 	public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
-		View view = inflater.inflate(R.layout.fragment_brick_add, container, false);
+		View view;
```

Comment:
dont do this.

this makes this a "tagged class", doing two completely different things with different layouts in a single class. the next guy will again just add behaviour here with help of some boolean flag.

https://books.google.at/books?id=ka2VUBqHiWkC&pg=PA101&lpg=PA101&dq=java++%22tagged+class%22&source=bl&ots=y_IoKgr_M4&sig=ACfU3U2qgbGTUfeD2HkpOgmn_FV7C6OCEA&hl=de&sa=X&ved=2ahUKEwj6h9mst-bkAhWEKVAKHeZSAgUQ6AEwB3oECAkQAQ#v=onepage&q=java%20%20%22tagged%20class%22&f=false

http://thefinestartist.com/effective-java/20

https://medium.com/@appmattus/effective-kotlin-item-23-prefer-class-hierarchies-to-tagged-classes-de99d37f815a

this should be a separate fragment.

Reply:
Just to make sure: You mean both fragments should inherit from ListFragment?
This would be a strong case of code duplication IMO.

Or did you mean to inherit from AddBrickFragment and just overwrite the onCreateView method?
Sorry for the inconvenience